src/sensor/parser.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <QObject>
+#include "ping-message.h"
+
+// This class digests data and notifies owner when something interesting happens
+class Parser : public QObject
+{
+    Q_OBJECT
+public:
+    Parser() = default;
+    ~Parser() = default;
+
+    uint32_t parsed; // number of messages/packets successfully parsed
+    uint32_t errors; // number of parse errors
+
+    // asynchronous use, Child classes should signal when something happens ie. 'emit newMessage(Message m)'
+    virtual void parseBuffer(const QByteArray& data) { Q_UNUSED(data) }
```

Comment:
This should be pure virtual

Reply:
I have a problem when I attempt to make it pure virtual:
![Screenshot_20190621_094559](https://user-images.githubusercontent.com/8315108/59926942-66b0f380-9409-11e9-93d0-973dab51330a.png)

Reply:
I assume it's because we have Parser member in abstract link, it the functions need to be defined.

Reply:
ok